department-of-veterans-affairs/caseflow
# hearing.date / hearing_day.hearing_date -> scheduled_for

Connects #8586 

### Description
This PR changes the date field on the hearing object and the hearing_date field on the hearing_day to scheduled_for on both AMA and legacy hearings.

It also adds some of the logic required to pull in AMA hearings on the upcoming hearings page in hearing prep. That will be continued in a future PR.

### Testing Plan
1. Confirm hearings are displayed correctly on the Case Details page
1. Confirm RO and judge algorithms run correctly
1. Confirm scheduling hearings works correctly

I merged a separate PR to make this migration backwards compatible.
